OpenAI-DotNet/Assistants/AssistantResponse.cs

```diff
@@ -95,8 +105,42 @@ public sealed class AssistantResponse : BaseResponse
         [JsonPropertyName("metadata")]
         public IReadOnlyDictionary<string, string> Metadata { get; private set; }
 
+        /// <summary>
+        /// What sampling temperature to use, between 0 and 2.
+        /// Higher values like 0.8 will make the output more random,
+        /// while lower values like 0.2 will make it more focused and deterministic.
+        /// </summary>
+        [JsonInclude]
+        [JsonPropertyName("temperature")]
+        public double Temperature { get; private set; }
+
+        /// <summary>
+        /// An alternative to sampling with temperature, called nucleus sampling,
+        /// where the model considers the results of the tokens with top_p probability mass.
+        /// So 0.1 means only the tokens comprising the top 10% probability mass are considered.
+        /// </summary>
+        [JsonInclude]
+        [JsonPropertyName("top_p")]
+        public double TopP { get; private set; }
+
+        /// <summary>
+        /// Specifies the format that the model must output.
+        /// Setting to <see cref="ResponseFormat.Json"/> enables JSON mode,
+        /// which guarantees the message the model generates is valid JSON.
+        /// </summary>
+        /// <remarks>
+        /// Important: When using JSON mode you must still instruct the model to produce JSON yourself via some conversation message,
+        /// for example via your system message. If you don't do this, the model may generate an unending stream of
+        /// whitespace until the generation reaches the token limit, which may take a lot of time and give the appearance
+        /// of a "stuck" request. Also note that the message content may be partial (i.e. cut off) if finish_reason="length",
+        /// which indicates the generation exceeded max_tokens or the conversation exceeded the max context length.
+        /// </remarks>
+        [JsonInclude]
+        [JsonPropertyName("response_format")]
+        public ResponseFormatObject ResponseFormat { get; private set; }
```

Comment:
Great work on this PR. I would love to assist where possible since I am using the SDK with the assistants v2 api.

It seems the deserialisation of this field fails in the assistant tests and also in a local package I build and tried to use. I dug in and tried to figure out how the string "response_format: value" is supposed to deserialise to this object and only came up with a custom converter to solve it and use the converter on this field. The error was basically "The JSON value could not be converted to OpenAI.ResponseFormatObject".

```
    public class ResponseFormatObjectConverter : JsonConverter<ResponseFormatObject>
    {
        public override ResponseFormatObject Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
        {
            var type = JsonSerializer.Deserialize<ResponseFormat>(ref reader, options);
            return new ResponseFormatObject(type);
        }

        public override void Write(Utf8JsonWriter writer, ResponseFormatObject value, JsonSerializerOptions options)
        {
            JsonSerializer.Serialize(writer, value.Type, options);
        }
    }
    
```
        [JsonInclude]
        [JsonPropertyName("response_format")]
        [JsonConverter(typeof(ResponseFormatObjectConverter))]
        public ResponseFormatObject ResponseFormat { get; private set; }
        
There are probably better ways, and perhaps other parts of code that are affected. Let me know if you want me to pitch in.

Reply:
Preliminary tests passed for me.

I'll look into incorporating something like this.

Reply:
Just thought I'd post an update here since I kept seeing weird behaviour. 

It seems the assistants api returns different data structures in the response_format. It can be both an object and a string. This is really over my head so I'll just wait for an official update :S

Reply:
I tested with JSON mode and didn't run into any trouble with my implementation in this PR

Reply:
I think the problem arises with "auto" mode for example. "Auto" is not part of `ResponseFormat` and is a string value in the response from the API. The json and text seem to be object values. The OpenAI Gui it self sets the value to "auto" when an assistant is created, so trying to fetch such an assistant does not work for me.

Reply:
> Just thought I'd post an update here since I kept seeing weird behaviour.
> 
> It seems the assistants api returns different data structures in the response_format. It can be both an object and a string. This is really over my head so I'll just wait for an official update :S

I found the same issue, when creating a new assistant via the api it is stored as

```
"response_format": {
        "type": "text"
      }
```

When I created an assistant via the openapi website it was stored as

`"response_format":"auto"`

I wrote an ugly converter that checks for `reader.TokenType == JsonTokenType.String` first, and in that case passes the value to the enum decoder. It appears to work in all cases I've tested it with so far.